### PR TITLE
fix:Settings tab should have 2 scrollable sections

### PR DESCRIPTION
**Pull request by Salman-Apptware**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)- https://linear.app/acryl-data/issue/PRD-742/settings-tab-should-have-2-scrollable-sections
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-web-react/src/app/entity/ownership/ManageOwnership.tsx [hunk 1]
```diff
@@ -20,7 +20,9 @@ const PageTitle = styled(Typography.Title)`
     }
 `;
 
-const ListContainer = styled.div``;
+const ListContainer = styled.div`
+    height: calc(100% - 120px);
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/ownership/ManageOwnership.tsx:24]: Salman - do you mind providing an overview of the changes in the PR description so I can better understand the rationale for solution?

File: datahub-web-react/src/app/entity/ownership/table/OwnershipTable.tsx [hunk 2]
```diff
@@ -55,6 +55,10 @@ export const OwnershipTable = ({ ownershipTypes, setIsOpen, setOwnershipType, re
                 emptyText: <Empty description="No Ownership Types found!" image={Empty.PRESENTED_IMAGE_SIMPLE} />,
             }}
             pagination={false}
+            style={{
+                height: 'calc(100% - 30px)',
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/ownership/table/OwnershipTable.tsx:59]: Why are we computing the height of the ownership table here? Can we try to use display: flex and flex-direction:column on the parent components for settings to see if the settings content can continue on down the page?

File: datahub-web-react/src/app/entity/view/ManageViews.tsx [hunk 3]
```diff
@@ -20,7 +20,9 @@ const PageTitle = styled(Typography.Title)`
     }
 `;
 
-const ListContainer = styled.div``;
+const ListContainer = styled.div`
+    height: calc(100% - 120px);
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/view/ManageViews.tsx:24]: Can you explain where 120px originates?

File: datahub-web-react/src/app/entity/view/ViewsTable.tsx [hunk 4]
```diff
@@ -56,6 +56,10 @@ export const ViewsTable = ({ views, onEditView }: ViewsTableProps) => {
                 emptyText: <Empty description="No Views found!" image={Empty.PRESENTED_IMAGE_SIMPLE} />,
             }}
             pagination={false}
+            style={{
+                height: 'calc(100% - 30px)',
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/view/ViewsTable.tsx:60]: Where does 30px originate?

File: datahub-web-react/src/app/identity/ManageIdentities.tsx [hunk 5]
```diff
@@ -27,7 +27,7 @@ const Content = styled.div`
         margin: 0;
     }
     color: #262626;
-    height: calc(100vh - 60px);
+    height: calc(100% - 60px);
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/identity/ManageIdentities.tsx:30]: where is 60px coming from?

File: datahub-web-react/src/app/identity/user/UserList.tsx [hunk 6]
```diff
@@ -25,10 +25,14 @@ import { useUpdateEducationStepIdsAllowlist } from '../../onboarding/useUpdateEd
...
 import { useUserContext } from '../../context/useUserContext';
 
-const UserContainer = styled.div``;
+const UserContainer = styled.div`
+    height: 100%;
+`;
 
 const UserStyledList = styled(List)`
     &&& {
+        height: calc(100% - 85px);
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/identity/user/UserList.tsx:34]: where does 85px come from?

File: datahub-web-react/src/app/permissions/policy/ManagePolicies.tsx [hunk 7]
```diff
@@ -38,7 +38,9 @@ import analytics, { EventType } from '../../analytics';
 import { POLICIES_CREATE_POLICY_ID, POLICIES_INTRO_ID } from '../../onboarding/config/PoliciesOnboardingConfig';
 import { OnboardingTour } from '../../onboarding/OnboardingTour';
 
-const SourceContainer = styled.div``;
+const SourceContainer = styled.div`
+    height: calc(100% - 40px);
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/permissions/policy/ManagePolicies.tsx:42]: where is 40px coming from?

File: datahub-web-react/src/app/permissions/roles/ManageRoles.tsx [hunk 8]
```diff
@@ -21,7 +21,9 @@ import { OnboardingTour } from '../../onboarding/OnboardingTour';
 import { ROLES_INTRO_ID } from '../../onboarding/config/RolesOnboardingConfig';
 import { clearUserListCache } from '../../identity/user/cacheUtils';
 
-const SourceContainer = styled.div``;
+const SourceContainer = styled.div`
+    height: calc(100% - 40px);
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/permissions/roles/ManageRoles.tsx:25]: where is 40px coming from?

File: datahub-web-react/src/app/search/SearchablePage.tsx [hunk 9]
```diff
@@ -138,7 +128,16 @@ export const SearchablePage = ({ onSearch, onAutoComplete, children }: Props) =>
                 authenticatedUserPictureLink={user?.editableProperties?.pictureLink}
                 entityRegistry={entityRegistry}
             />
-            <div style={styles.children}>{children}</div>
+            <div
+                style={{
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/search/SearchablePage.tsx:132]: where does this 60px coming from?

**Review by jjoyce0510 — COMMENTED** [review]
Overall I am a bit concerned about the level of hardcoding pixels we have in this one. I was hoping we'd be able to achieve this by examining the layout of the settings page, with the expectation that the right content view would occupy 100% of the screen view (full screen), while the left side of the settings page could be independently scrollable.. 

Is there any way we can try to reduce the specific pixel constraints here?

File: datahub-web-react/src/app/entity/ownership/ManageOwnership.tsx (hunk 1 above)
Comment (Salman-Apptware) [reply datahub-web-react/src/app/entity/ownership/ManageOwnership.tsx:24]: I have calculate PageHeaderContainer height with inspecting in browser i.e 120px and trying to remove from ListContainer to make it scrollable. I have check on different resoltion like 1920, 1320 it was properly. but if pixel is concern let me try with vh or Percentage

Comment (Salman-Apptware) [reply datahub-web-react/src/app/entity/ownership/ManageOwnership.tsx:24]: can be replace with height: calc(100% - 11vh); after calculating PageHeaderContainer and Pagination

File: datahub-web-react/src/app/entity/ownership/table/OwnershipTable.tsx (hunk 2 above)
Comment (Salman-Apptware) [reply datahub-web-react/src/app/entity/ownership/table/OwnershipTable.tsx:59]: after setting  height: calc(100% - 11vh);  to MangeOwnerShip.tsx notneed to add height just overflow can add the scroll.

File: datahub-web-react/src/app/entity/view/ManageViews.tsx (hunk 3 above)
Comment (Salman-Apptware) [reply datahub-web-react/src/app/entity/view/ManageViews.tsx:24]: can be replace with height: calc(100% - 11vh); after calculating PageHeaderContainer and Pagination

File: datahub-web-react/src/app/entity/view/ViewsTable.tsx (hunk 4 above)
Comment (Salman-Apptware) [reply datahub-web-react/src/app/entity/view/ViewsTable.tsx:60]: after setting height: calc(100% - 11vh); to MangeView.tsx not need to add height just overflow can add the scroll.

File: datahub-web-react/src/app/search/SearchablePage.tsx (hunk 9 above)
Comment (Salman-Apptware) [reply datahub-web-react/src/app/search/SearchablePage.tsx:132]: header(top nav bar) is 60px

File: datahub-web-react/src/app/permissions/policy/ManagePolicies.tsx (hunk 7 above)
Comment (Salman-Apptware) [reply datahub-web-react/src/app/permissions/policy/ManagePolicies.tsx:42]: make PageContainer height: calc(100% - 11vh);  SourceContainer height: 100%

File: datahub-web-react/src/app/permissions/roles/ManageRoles.tsx (hunk 8 above)
Comment (Salman-Apptware) [reply datahub-web-react/src/app/permissions/roles/ManageRoles.tsx:25]: make PageContainer height: calc(100% - 11vh);  SourceContainer height: 100%

**Comment by Salman-Apptware** [comment]
I have remove px and add vh according to resolution. This is temporary solution but we have to take each component and resolve structural changes in new tickets

File: datahub-web-react/src/app/entity/ownership/ManageOwnership.tsx [hunk 10]
```diff
@@ -20,7 +20,9 @@ const PageTitle = styled(Typography.Title)`
     }
 `;
 
-const ListContainer = styled.div``;
+const ListContainer = styled.div`
+    height: calc(100% - 11vh);
+`;
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/entity/ownership/ManageOwnership.tsx:25]: this should not be needed. please use flexbox for making height dynamic.

**Comment by gabe-lyons** [comment]
@Salman-Apptware vh is not the correct approach here. You should not need to do any height calculations at all. This should be handled dynamically by flexbox.

**Comment by Salman-Apptware** [comment]
closing this PR and created new one with flex box

**Pull request closed without merging**